### DISCUSSION TITLE
Unable to obtain response header information

### DISCUSSION THREAD

**Discussion by W2426566602** [reactiveui/refit#1698] [Q&A] [answered]
When I return a FileResult in the API, I use the browser developer tools to view the request details, view the request response, and there is indeed a head information in the response, but when I use refit to get the response, I find that there is no head information

**Comment by glennawatson** [comment, marked as answer]
To read response headers with Refit, change the method return type so Refit hands you the response, not just the deserialized body. By default Refit returns only the body, so headers are discarded.

Use one of:

- Task<HttpResponseMessage> - full raw response, read resp.Headers and resp.Content.Headers yourself.
- Task<IApiResponse<T>> (or Task<ApiResponse<T>>) - gives you the deserialized body plus .Headers and .ContentHeaders.

```csharp
[Get("/download")]
Task<HttpResponseMessage> Download();

// or
[Get("/download")]
Task<IApiResponse<Stream>> Download();
```

Then:

```csharp
var resp = await api.Download();
var disposition = resp.Content.Headers.ContentDisposition;       // for HttpResponseMessage
// or
var disposition = resp.ContentHeaders?.ContentDisposition;       // for IApiResponse<T>
```

Note that some headers (like Content-Disposition, Content-Type, Content-Length) live on Content.Headers / ContentHeaders, not the top-level response Headers, which is why a plain body return type never exposes them.
